proposals/dictionary-expressions.md

```diff
@@ -130,41 +184,46 @@ struct Pair<X, Y>
 
 Dictionary<int, string> map1 = [pair1, pair2]; // ?
 
-
 List<Pair<int, string>> pairs = ...;
 Dictionary<int, string> map2 = [.. pairs]; // ?
 ```
 
-### Open question 5
-
-Dictionaries provide two ways of initializing their contents.  A restrictive `.Add`-oriented form that throws when a key is already present in the dictionary, used by collection initializers, and a permissive indexer-oriented form which does not, used by dictionary initializers.  The restrictive form is useful for catching mistakes ("oops, I didn't intend to add the same thing twice!"), but is limiting *especially* in the spread case.  For example:
+Similarly, instead of `KeyValuePair<,>` we could allow *any* type deconstructible to two values? For example:
 
 ```c#
-Dictionary<string, Option> optionMap = [opt1Name: opt1Default, opt2Name: opt2Default, .. userProvidedOptions];
+record struct Pair<X, Y>(X x, Y y);
+
+Dictionary<int, string> map1 = [pair1, pair2]; // ?
 ```
 
-Or, conversely:
+Resolution: TBD.  Working group recommendation: Only allow `KeyValuePair<,>` for now.  Do not do anything with tuples and/or other deconstructible types.  This is also something that could be relaxed later if there is sufficient motivation.
 
-```c#
-Dictionary<string, Option> optionMap = [.. Defaults.CoreOptions, feature1Name: feature1Override];
-```
+### Question: Types that support both collection and dictionary initialization
 
-Which approach should we go with with our dictionary expressions? Options include:
+We will support *key value pair elements* for C#12 collection expression target types. Which approach should we use for initialization for those types?
```

Comment:
From the example and options below, it sounds like the question is about valid C#12 syntax rather than `key_value_pair_element`. Consider clarifying, perhaps:
```suggestion
C# 12 supports collection types where the element type is some `KeyValuePair<,>`, where the type has an applicable `Add()` method that takes a single argument. Which approach should we use for initialization if the type also includes an indexer?
```